src/datasets/utils/patching.py

```diff
@@ -0,0 +1,91 @@
+import importlib
+import os
+from contextlib import contextmanager
+
+from .logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class _PatchedModuleObj:
+    """Set all the modules components as attributes of the _PatchedModuleObj object"""
+
+    def __init__(self, module):
+        if module is not None:
+            for key in getattr(module, "__all__", module.__dict__):
+                if not key.startswith("__"):
+                    setattr(self, key, getattr(module, key))
+
+
+class patch_submodule:
+    """
+    Patch a submodule attribute of an object, by keeping all other submodules intact at all levels.
+
+    Example::
+
+        >>> import importlib
+        >>> from datasets.load import prepare_module
+        >>> from datasets.streaming import patch_submodule, xjoin
+        >>>
+        >>> snli_module_path, _ = prepare_module("snli")
+        >>> snli_module = importlib.import_module(snli_module_path)
+        >>> patcher = patch_submodule(snli_module, "os.path.join", xjoin)
+        >>> patcher.start()
+        >>> assert snli_module.os.path.join is xjoin
```

Comment:
Totally unrelated to the PR, but why the `>>>` prefix for example? It makes it impossible to copy-n-paste the code. I know pytorch uses this style and the code examples in their docs are very hard to use because of that.

The `::` already ensures that sphinx will render this section as a code segment.

Reply:
No longer applicable to this PR once that I have followed another approach different from patching. Anyway, I take on this in another PR.

Reply:
I know the `>>>` prefix is normally used, so that the examples are tested with [`doctest`](https://docs.python.org/3/library/doctest.html). However I do not think testing is intended for this examples in this docstring... So I agree it would be better to remove the prefix.

Reply:
Thank you for validating that, @albertvillanova!

So it only impacts those of us who read the source code.